File: Solutions/arpitmisraw/Task5/threading.py
import threading

def factorial(n):
    i = 1
    ans = 1
    while(i <= n):
        ans *= i
        i += 1
    print("Factorial of", n, "is", ans)


def square(n):
    print("Square of", n, "is", n**2)


n = int(input())
t1 = threading.Thread(target=factorial, args=(n,))
t2 = threading.Thread(target=square, args=(n,))
t1.start()
t2.start()
t1.join()
t2.join()

File: Solutions/marshallKR7/Task-2/Threading.cpp
#include<iostream.h>
#include<thread.h>
using namespace std;

void fact(long long int n)
{
	long long int ans=1,i;
	for(i=2;i<=n;i++)
	{
		ans=(ans*i)%1000000007;
	}
	cout<<"The factorial is"<<ans<<"\n";
}

void expo(long long int t)
{
	long long int base,exp,final=1;
	base=t;
	exp=t;
	while(exp>0)
	{
		if(exp%2!=0)
			final=(final*base)%1000000007;
		base=(base*base)%1000000007;
		exp/=2;
	}
	cout<<"The exponential of number with respect to itself is"<<final<<"\n";
}

int main()
{
	long long int a;
	cin>>a;
    thread t1(&fact,a);
    thread t2(&expo,a);
    t1.join();
    t2.join();
	return 0;
}
